src/rust/engine/fs/store/src/lib.rs

```diff
@@ -427,7 +427,7 @@ impl Store {
   /// two functions f_local and f_remote. These functions are any validation or transformations you
   /// want to perform on the bytes received from the local and remote cas (if remote is configured).
   ///
-  async fn load_bytes_with<
+  pub async fn load_bytes_with<
```

Comment:
@stuhood  / @illicitonion: Should I add a `Tree`-specific helper instead of making this method public? What should that method look like?

Reply:
This method is called by two other methods in this impl: `load_file_bytes_with` and `load_directory`... you probably want to mimic the latter one rather than exposing this method.

One thing to note though is that the local and remote stores do not actually have a shared interface... that was supposed to provide flexibility for this kind of situation. If you can imagine a way for the local store to not store trees, and to have that be a remote-only concern, that would be great... if on the other hand it looks simpler to just continue to have a 1:1 mapping, that's fine too.

Reply:
Maybe a `load_tree_from_remote` method that could load from the remote CAS (since it would have access to Store.remote) and just insert the embedded Directory protos into the LMDB store without storing the Tree protos locally?

> One thing to note though is that the local and remote stores do not actually have a shared interface

Then would it make sense to remove parameters (like `entry_type`) that are unused in the remote interface?

Reply:
Yea, I think that that would make sense.